nucypher/network/protocols.py

```diff
@@ -9,17 +11,16 @@ class SuspiciousActivity(RuntimeError):
     """raised when an action appears to amount to malicious conduct."""
 
 
-def parse_node_uri(uri: str):
+def parse_node_uri(uri: str, delimiter: str = '@') -> Tuple[str, int, Address]:
     from nucypher.config.characters import UrsulaConfiguration
 
-    if '@' in uri:
-        checksum_address, uri = uri.split("@")
+    checksum_address = None
+    if delimiter in uri:
+        checksum_address, uri = uri.split(delimiter)
         if checksum_address is None:
             raise ValueError(f"{uri} is not a valid Teacher URI - no checksum address.")
         if not is_checksum_address(checksum_address):
             raise ValueError("{} is not a valid checksum address.".format(checksum_address))
-    else:
```

Comment:
Should we raise a value error if delimiter not in uri?

Reply:
No, because the delimiters presence is optional and this function parses URLs with or without it.  I'm unsure if it's being used anymore, but theoretically peers can be verified and filtrated by a checksum address.